src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -350,51 +350,42 @@ private bool TryParseNamedEnum(
 
         private static ulong ConvertToUInt64(T value)
         {
-            switch (s_enumTypeCode)
-            {
-                case TypeCode.Int32 or TypeCode.UInt32: return Unsafe.As<T, uint>(ref value);
-                case TypeCode.Int64 or TypeCode.UInt64: return Unsafe.As<T, ulong>(ref value);
-                case TypeCode.Int16 or TypeCode.UInt16: return Unsafe.As<T, ushort>(ref value);
-                default:
-                    Debug.Assert(s_enumTypeCode is TypeCode.SByte or TypeCode.Byte);
-                    return Unsafe.As<T, byte>(ref value);
+            return s_enumTypeCode switch
+            {
+                TypeCode.Int32 => (ulong)(int)(object)value,
+                TypeCode.UInt32 => (uint)(object)value,
+                TypeCode.Int64 => (ulong)(long)(object)value,
+                TypeCode.UInt64 => (ulong)(object)value,
+                TypeCode.Int16 => (ulong)(short)(object)value,
+                TypeCode.UInt16 => (ushort)(object)value,
+                TypeCode.SByte => (ulong)(sbyte)(object)value,
+                _ => (byte)(object)value
             };
         }
 
         private static long ConvertToInt64(T value)
         {
-            Debug.Assert(s_isSignedEnum);
-            switch (s_enumTypeCode)
-            {
-                case TypeCode.Int32: return Unsafe.As<T, int>(ref value);
-                case TypeCode.Int64: return Unsafe.As<T, long>(ref value);
-                case TypeCode.Int16: return Unsafe.As<T, short>(ref value);
-                default:
-                    Debug.Assert(s_enumTypeCode is TypeCode.SByte);
-                    return Unsafe.As<T, sbyte>(ref value);
+            return s_enumTypeCode switch
+            {
+                TypeCode.Int32 => (int)(object)value,
+                TypeCode.Int64 => (long)(object)value,
+                TypeCode.Int16 => (short)(object)value,
+                _ => (sbyte)(object)value,
```

Comment:
Is there inconsistency here? Some methods are expecting unsigned type in `s_enumTypeCode`, while this is not.

Reply:
Oops, accidentally removed `Debug.Assert(s_isSignedEnum);` assert.

Reply:
> Oops, accidentally removed Debug.Assert(s_isSignedEnum); assert.

Are you going to put it back?

Reply:
> > Oops, accidentally removed Debug.Assert(s_isSignedEnum); assert.
> 
> Are you going to put it back?

Addded